drivers/flash/spi_nor.h

```diff
@@ -7,8 +7,16 @@
 #ifndef __SPI_NOR_H__
 #define __SPI_NOR_H__
 
+#include <zephyr/drivers/flash.h>
 #include <zephyr/sys/util.h>
 
+enum flash_spi_nor_ops {
+	/*
+	 * Sends Flash reset SPI commands to reset the flash device.
+	 */
+	FLASH_SPI_NOR_RESET = FLASH_EX_OP_VENDOR_BASE,
```

Comment:
I would rather go with base and define this as FLASH_RESET command. Not all devices have to support it but that is something that will be quite universal.
This is reset to init state?

Reply:
yes this is to reset to initial state. 
I see the reset commands are part of the spi_nor.h i.e commands below.

#define SPI_NOR_CMD_RESET_EN    0x66    /* Reset Enable */
#define SPI_NOR_CMD_RESET_MEM   0x99    /* Reset Memory */

This implementation sends above two commands only nothing additionally added.
If the flash chip do not support, applications should not enable CONFIG_FLASH_EX_OP_ENABLED.

Please let me know if i am missing anything here.

Reply:
> yes this is to reset to initial state. I see the reset commands are part of the spi_nor.h i.e commands below.
> 
> #define SPI_NOR_CMD_RESET_EN 0x66 /* Reset Enable _/ #define SPI_NOR_CMD_RESET_MEM 0x99 /_ Reset Memory */
> 
> This implementation sends above two commands only nothing additionally added. If the flash chip do not support, applications should not enable CONFIG_FLASH_EX_OP_ENABLED.
> 

That is correct if you are not going to send ex_ops you are not going to enabled that. ex_op is provided to avoid growing the flash_driver_api for every function there is and just keep the limited number of direct callbacks, and keep all the non-essential things out of the api. It basically drops responsibility of recognizing what extras are supported to the driver, in ex_op code.

> Please let me know if i am missing anything here.

The vendor space is reserved for vendor characteristic functions, like protections in stm32 that do not appear in other chips. 

The reset on the other side may be something that can be provided by many chips, with no specific vendor required.
So I guess we now need first official ex op like FLASH_EX_OP_REINIT_DEVICE or FLASH_EX_OP_RESET_COMMAND that will indicate that the flash will be requested to perform reset in software rather than by pulling some leg up or down.

Such command would not be based in _VENDOR_BASE but rather have first available ID, here it would be 0.

Reply:
The definition should not mention type of device, like _SPI_NOR_ in this case, and should be moved, as it will be systemwide, to the [include/zephyr/drivers/flash.h](https://github.com/zephyrproject-rtos/zephyr/blob/69fd128771e3d13933254f46eda88d7a715d8a44/include/zephyr/drivers/flash.h#L471)

Reply:
Thanks, moved it to flash.h